mep-0002-simplify-data-structure.md

```diff
@@ -0,0 +1,392 @@
+        super().__init__(*args, **kwargs)
+
+
+class TensorChunkData(EntityData):
+    _shape = TupleField("shape", FieldTypes.int64)
+    _order = ReferenceField("order", TensorOrder)
+    _index = TupleField("index", FieldTypes.uint32)
+
+    def __init__(self, op=None, index=None, shape=None, dtype=None, order=None, **kw):
+        args = ()
+        kwargs = {
+            "_shape": shape,
+            "_order": order,
+            "_op": op,
+            "_index": index,
+            "_extra_params": {"_i": kw["_i"]},
+            "_key": kw["_key"],
+            "_dtype": kw.get("_dtype"),
+            "_id": str(id(self)),
+        }
+        super().__init__(*args, **kwargs)
+
+```
+
+We constructed a `TensorChunkData`:
+
+![mep-0002_08](https://user-images.githubusercontent.com/5388750/231441726-80de2790-29c9-4666-b6de-3315ad13ebac.png)
+
+The cost is about 3.55e-06s.
+
+Secondly, we define a `TensorChunkData` which extends `Serializable` directly.
+
+```python
+class TensorChunkData(Serializable):
+    _shape = TupleField("shape", FieldTypes.int64)
+    _order = ReferenceField("order", TensorOrder)
+    _index = TupleField("index", FieldTypes.uint32)
+    _extra_params = DictField("extra_params", key_type=FieldTypes.string)
+    _key = StringField("key", default=None)
+    _id = StringField("id")
+
+    def __init__(self, op=None, index=None, shape=None, dtype=None, order=None, **kw):
+        args = ()
+        kwargs = {
+            "_shape": shape,
+            "_order": order,
+            "_op": op,
+            "_index": index,
+            "_extra_params": {"_i": kw["_i"]},
+            "_key": kw["_key"],
+            "_dtype": kw.get("_dtype"),
+            "_id": str(id(self)),
+        }
+        super().__init__(*args, **kwargs)
+```
+
+We constructed a `TensorChunkData`:
+
+![mep-0002_09](https://user-images.githubusercontent.com/5388750/231441728-af08d374-d327-45c0-803d-c611f4dda7ad.png)
+
+The cost is about 2.81e-06s, which is reduced by **20.8%**.
+
+There is another way to alleviate time-consuming effect of **5**.
+
+Firstly, we define a `TensorChunk`:
+
+```python
+class TensorChunk(Serializable):
+    type_name = "Tensor"
+    _data = ReferenceField("data", EntityData)
+
+    def __init__(self, data, **kwargs):
+        super().__init__(_data=data, **kwargs)
+```
+
+We constructed a `TensorChunk`:
+
+![mep-0002_10](https://user-images.githubusercontent.com/5388750/231441729-22081b47-241a-4944-a283-70ae34795194.png)
+
+The cost is about 8.28e-07s which accounts for **29.5%** of constructing a `TensorChunkData`.
+We can largely alleviate this kind of time-consuming problem by merging `Chunk` and `ChunkData` into `Chunk`.
+
+### Modification of Operand
+
+We need to generate a `op_id` field for every Operand, and maintain a mapping from `op_id` to Operand as follows to find the corresponding Operand and construct the Operand instance.
+For convenience, we take the following steps to generate `op_id` and mapping:
```

Comment:
Can we use opcodes in `opcodes.py` instead and setup mapping between opcodes and operators?

Reply:
Looks like the opcode is useless. The opcode is used as the `_op_type_` field of `Operand`. But some operands do not have this field, like `DataFrameGetDummies`; some operands have the same field, like `LearnShuffleProxy`, `TensorShuffleProxy`, and `DataFrameShuffleProxy`; some operands' `_op_type_` are `None`, like `TensorCpFuseChunk` and `TensorNeFuseChunk`.